appinventor/components/src/com/google/appinventor/components/scripts/ComponentProcessor.java

```diff
@@ -1944,6 +1956,39 @@ private Option elementToOption(Element field, String value) {
     );
   }
 
+  /**
+   * Returns the associated helper key if the element has an @Asset annotation. Null otherwise.
+   *
+   * @param elem the Element which represents a function (for return types) or a parameter.
+   * @param type the TypeMirror representing the type of that element.
+   * @return the associated helper key if the element has an @Asset annotation.
+   */
+  private HelperKey hasAssetsHelper(Element elem, TypeMirror type) {
+    for (AnnotationMirror mirror : elem.getAnnotationMirrors()) {
+      if (mirror.getAnnotationType().asElement().getSimpleName().contentEquals("Asset")) {
+        int index = 0;  // Index 0 is the empty filter.
+        for(Entry<? extends ExecutableElement, ? extends AnnotationValue> entry:
+            mirror.getElementValues().entrySet()) {
+          // Make sure we are looking at the value attribute.
+          if (!entry.getKey().getSimpleName().contentEquals("value")) {
+            continue;
+          }
+          List<AnnotationValue> values = (List<AnnotationValue>)entry.getValue().getValue();
+          List<String> filter = new ArrayList<String>();
+          for (AnnotationValue v : values) {
+            filter.add((String)v.getValue());
```

Comment:
I think we should also toLowerCase and sort the filter values. Otherwise, the following annotations are different:

1. `@Assets({".ext1", ".ext2"})`
2. `@Assets({".ext2", ".ext1"})`
3. `@Assets({".EXT1", ".EXT2"))`
4. `@Assets({".EXT2", ".EXT1"))`